src/pivot-table/components/cells/styles.tsx

```diff
@@ -1,21 +1,31 @@
 import Box from "@mui/material/Box";
 import { styled } from "@mui/material/styles";
 import type { stardust } from "@nebula.js/stardust";
+import { textStyle } from "../shared-styles";
 
-interface Props {
+const baseFlex: React.CSSProperties = {
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+};
+
+interface StyledHeaderCellWrapperProps {
   interactions: stardust.Interactions;
   background: string;
   hoverBackground: string;
+  shouldShowMenuIcon: boolean;
 }
 
 export const StyledHeaderCellWrapper = styled(Box, {
-  shouldForwardProp: (prop: string) => prop !== "interactions" && prop !== "hoverBackground" && prop !== "background",
-})(({ interactions, background, hoverBackground }: Props) => ({
-  padding: 0,
+  shouldForwardProp: (prop: string) =>
+    !["interactions", "hoverBackground", "background", "shouldShowMenuIcon"].includes(prop),
+})(({ interactions, background, hoverBackground, shouldShowMenuIcon }: StyledHeaderCellWrapperProps) => ({
+  // inline styling uses getBorderStyle(..) which has a padding of 4px and it overrides this
+  padding: "0 !important",
```

Comment:
suggestions are welcome on this one to eliminate `!important`

Reply:
I think you can get rid of it and also properly align the header cell and the dimension cells below it.

It should look something like this:
![Skärmavbild 2023-10-02 kl  11 00 34](https://github.com/qlik-oss/sn-pivot-table/assets/16608020/703dc8bb-07e0-445d-a3a9-15437c6b1737)
![Skärmavbild 2023-10-02 kl  11 03 21](https://github.com/qlik-oss/sn-pivot-table/assets/16608020/18032fcd-c664-438c-9f5b-cfa492ee7f3d)


But currently looks like this:
![Skärmavbild 2023-10-02 kl  11 02 37](https://github.com/qlik-oss/sn-pivot-table/assets/16608020/46ff6a88-d857-42b9-b778-a1e5fa67ee97)
![Skärmavbild 2023-10-02 kl  11 02 58](https://github.com/qlik-oss/sn-pivot-table/assets/16608020/0e6c664e-fdfc-4979-97f0-33711e30854f)

Reply:
Example code below, but I've not yet tested every scenerio:
```ts
export const StyledHeaderCellWrapper = styled("div", {
  shouldForwardProp: (prop: string) =>
    !["interactions", "hoverBackground", "background", "shouldShowMenuIcon"].includes(prop),
})(({ interactions, background, hoverBackground, shouldShowMenuIcon }: StyledHeaderCellWrapperProps) => ({
  position: "relative",
  display: "grid",
  gridTemplateColumns: shouldShowMenuIcon ? "1fr 24px" : "1fr",
  gridGap: shouldShowMenuIcon ? "4px" : "0px",
  alignItems: "center",
  cursor: interactions.active ? "pointer" : "default",
  background,
  "&&:hover": {
    background: interactions.active ? hoverBackground : background,
  },
}));

export const StyledHeaderAnchor = styled(Box)({
  position: "absolute",
  left: 0,
  bottom: 0,
});

export const StyledHeaderCell = styled("div")(() => ({
  ...baseFlex,
  flexDirection: "row",
  overflow: "hidden",
  position: "relative",
  gap: "4px",
}));

export const StyledLockIcon = styled("div")(() => ({
  ...baseFlex,
}));

interface StyledLabelProps {
  shouldShowMenuIcon: boolean;
  fontSize: string;
  fontFamily: string;
}

export const StyledLabel = styled("div", {
  shouldForwardProp: (prop: string) => !["shouldShowMenuIcon", "fontSize", "fontFamily"].includes(prop),
})(({ fontSize, fontFamily }: StyledLabelProps) => ({
  ...textStyle,
  fontWeight: "600",
  alignSelf: "center",
  flexGrow: 1,
  fontSize,
  fontFamily,
}));
```